cms/static/cms/js/modules/cms.structureboard.js

```diff
@@ -1070,12 +1068,17 @@ class StructureBoard {
         const placeholderIds = getPlaceholderIds(CMS._plugins).map(id => `placeholders[]=${id}`).join('&');
 
         return $.ajax({
-            url: Helpers.updateUrlWithPath(
+            url: (
```

Comment:
What were you thoughts when removing: `Helpers.updateUrlWithPath`?

Reply:
`Helpers.updateUrlWithPath` is the root caues of the bug. It replaces query separator`&` by `&amp`s. Since it is used in many places within django CMS, I did not change it. I cannot anticipate the side effects. 

In any case: `Helpers.updateUrlWithPath('/local_url?x=1&y=2')` returns `'/local_url?x=1&amp;y=2&amp;cms_path=%2Fadmin%2Fcms%2Fplaceholder%2Fobject%2F10%2F'` which is not a meanigful path to get the toolbar for. The fix in the same situation creates a meaningful url: `'/local_url?x=1&y=2&cms_path=%2Fadmin%2Fcms%2Fplaceholder%2Fobject%2F10%2F'`

Frankly, I still struggle to comprehend the implementation of `Helpers.makeUrl` which is called by `Helpers.updateUrlWithPath`: https://github.com/django-cms/django-cms/blob/914558d283c197b4035ae7e1a084860f486c9429/cms/static/cms/js/modules/cms.base.js#L291-L308

Reply:
An alternative way would be to drop the `urljs` dependence and use the browsers' `URLSearchParams` interface. This would allow to have `Helpers.updateUrlWithPath` to be present. It should build correct urls. But it also requires a small change in `cms.pagetree.js`.  Overall this is probably cleaner but a be it small change deep in the code.

If you think that is better, I'll give it a go!

Reply:
After more pondering I come to the conclusion to essentially remove lines 303-307 in `makeURL`:  It seems to never have worked right with more than one query parameter. Overall, this deems to be the most systematic solution. 

I still have to make the discrimination test for `Resolver404` work.